src/Shared/LayoutRenderers/IISInstanceNameLayoutRenderer.cs

```diff
@@ -32,28 +33,45 @@ namespace NLog.Web.LayoutRenderers
     public class IISInstanceNameLayoutRenderer : LayoutRenderer
     {
 #if ASP_NET_CORE
+        private IHostEnvironment RetrieveHostEnvironment()
```

Comment:
Could this method be moved down below `HostEnvironment`-property, and be changed to this:
```c#
private IHostEnvironment RetrieveHostEnvironment()
{
#if ASP_NET_CORE
    return ServiceLocator.ResolveService<IHostEnvironment>(ResolveService<IServiceProvider>(), LoggingConfiguration);
#else
    return Internal.HostEnvironment.Default
#endif
}
```